addons/point_of_sale/models/pos_config.py

```diff
@@ -720,23 +720,43 @@ def _link_same_non_cash_payment_methods(self, source_config):
         if pms:
             self.payment_method_ids = [Command.link(pm.id) for pm in pms]
 
-    def _ensure_cash_payment_method(self, journal_code, name):
-        self.ensure_one()
-        if not self.company_id.chart_template or self.payment_method_ids.filtered('is_cash_count'):
-            return
-        company_id = self.company_id.id
-        cash_journal = self.env['account.journal'].create({
+    def _is_journal_exist(self, journal_code, name, company_id):
+        account_journal = self.env['account.journal']
+        existing_journal = account_journal.search([
+            ('name', '=', name),
+            ('code', '=', journal_code),
+            ('company_id', '=', company_id),
+        ], limit=1)
+
+        return existing_journal.id or account_journal.create({
             'name': name,
             'code': journal_code,
             'type': 'cash',
             'company_id': company_id,
-        })
-        cash_pm = self.env['pos.payment.method'].create({
+        }).id
+
+    def _is_pos_cash_pm_exist(self, name, journal_id, company_id):
```

Comment:
Is this really only for cash payment method ? Maybe rename it `_is_pos_pm_exist`